docs/process/labels.md

```diff
@@ -0,0 +1,121 @@
+# Issue and Pull Request Labels
+
+This section outlines the labels currently used in the project, and adds context
+about what each label represents.
+
+## Common labels
+
+These labels are used for both issues and pull requests, and are assigned to
+help understand the type of work involved.
+
+|                               | Label name       | Description |
+| ----------------------------- | -----------------| ----------- |
+| [:mag_right:][docs]           | `docs`           | Issues and pull requests related to documentation work on the project |
+| [:mag_right:][infrastructure] | `infrastructure` | Issues and pull requests related to scripts and tooling for GitHub Desktop |
+| [:mag_right:][tech-debt]      | `tech-debt`      | Issues and pull requests related to addressing technical debt or improving the codebase |
+
+## Issue-specific labels
+
+This section is organized into sub-groups, as groups of labels relate to
+specific work being done in the issue tracker.
+
+### Issue triage
+
+The triage process is how the maintainers process incoming issues and prioritize
+the work required to address the feedback raised. These labels help us to track
+the flow of an issue through this process
+
+|                                        | Label name                  | Description |
+| -------------------------------------- | ----------------------------| ----------- |
+| [:mag_right:][bug]                     | `bug`                     | Confirmed bugs or reports that are very likely to be bugs |
+| [:mag_right:][enhancement]             | `enhancement`             | |
+| [:mag_right:][investigation-needed]    | `investigation-needed`    | Likely bugs, but haven't been reliably reproduced by a reviewer |
+| [:mag_right:][more-information-needed] | `more-information-needed` | The submitter needs to provide more information about the issue |
+| [:mag_right:][priority-1]              | `priority-1`              | Major bug affecting large population and inhibiting their work |
+| [:mag_right:][priority-2]              | `priority-2`              | Bug that affects more than a few users in a meaningful way but doesn't prevent core functions |
+| [:mag_right:][priority-3]              | `priority-3`              | Bugs that affect small number of users and/or relatively cosmetic in nature |
+
+### External contributions
+
+We use these labels to identify work that is ideal for external contributors to
+get involved with.
+
+|                                 | Label name         |  Description |
+| ------------------------------- | ------------------ |  ----------- |
+| [:mag_right:][good first issue] | `good first issue` | Issues marked as ideal for a brand new contributor to start with |
```

Comment:
Issues is a touch redundant here

Reply:
I'm still somewhat attached to the "Issues" prefix here to differentiate them from "Issues and pull requests ..." and now that you point this out, those places that we aren't following that convention now stand out that. If someone with stronger opinions wants to have a go at the verbiage after this PR, they can be my guest.